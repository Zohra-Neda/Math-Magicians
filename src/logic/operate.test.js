/*eslint-disable*/
import operate from './operate';
import Big from 'big.js';
test('should add two numbers correctly', () => {
    const result = operate(2, 3, '+');
    expect(result).toMatchSnapshot();
  });
  
  test('should subtract two numbers correctly', () => {
    const result = operate(5, 2, '-');
    expect(result).toMatchSnapshot();
  });
  
  // Add more test cases for other operations
  
  // Test error cases
  test('should handle division by zero', () => {
    const result = operate(10, 0, 'รท');
    expect(result).toMatchSnapshot();
  });
  
  test('should handle modulo with zero divisor', () => {
    const result = operate(10, 0, '%');
    expect(result).toMatchSnapshot();
  });