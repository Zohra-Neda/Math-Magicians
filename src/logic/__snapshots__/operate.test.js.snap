// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should add two numbers correctly 1`] = `"5"`;

exports[`should handle division by zero 1`] = `"Can't divide by 0."`;

exports[`should handle modulo with zero divisor 1`] = `"Can't find modulo as can't divide by 0."`;

exports[`should subtract two numbers correctly 1`] = `"3"`;
